src/main/scala/uk/gov/nationalarchives/tdr/api/utils/TreeNodesUtils.scala

```diff
@@ -0,0 +1,36 @@
+package uk.gov.nationalarchives.tdr.api.utils
+
+import java.io.{File => JIOFile}
+import java.util.UUID
+import uk.gov.nationalarchives.tdr.api.model.file.NodeType.{fileTypeIdentifier, folderTypeIdentifier}
+import uk.gov.nationalarchives.tdr.api.service.UUIDSource
+import uk.gov.nationalarchives.tdr.api.utils.TreeNodesUtils.TreeNode
+
+import scala.annotation.tailrec
+
+class TreeNodesUtils(uuidSource: UUIDSource) {
+  def generateNodes(filePaths: Set[String]): Map[String, TreeNode] = {
+    @tailrec
+    def innerFunction(originalPath: String, fileType: String, nodes: Map[String, TreeNode]): Map[String, TreeNode] = {
+      val jioFile = new JIOFile(originalPath)
+      val parentPath = Option(jioFile.getParent)
```

Comment:
You're correct that having a leading `/` in the file path causes an entry without a `Name`. However, I don't think the client side files paths would include a leading `/`. That said I wondering whether we should be cleaning the file path string to remove leading `/` to prevent this from happening.